src/aof.c

```diff
@@ -40,191 +40,603 @@
+void aofInfoFree(aofInfo *ai) {
+    serverAssert(ai != NULL);
+    if (ai->file_name) sdsfree(ai->file_name);
+    zfree(ai);
+}
+
+/* Deep copy an aofInfo. */
+aofInfo *aofInfoDup(aofInfo *orig) {
+    serverAssert(orig != NULL);
+    aofInfo *ai = aofInfoCreate();
+    ai->file_name = sdsdup(orig->file_name);
+    ai->file_seq = orig->file_seq;
+    ai->file_type = orig->file_type;
+    return ai;
+}
+
+/* Method to free AOF list elements. */
+void aofListFree(void *item) {
+    aofInfo *ai = (aofInfo *)item;
+    aofInfoFree(ai);
+}
+
+/* Method to duplicate AOF list elements. */
+void *aofListDup(void *item) {
+    return aofInfoDup(item);;
+}
+
+/* Create an empty aofManifest, which will be called in `loadAofManifestFromDisk`. */
+aofManifest *aofManifestCreate(void) {
+    aofManifest *am = zcalloc(sizeof(aofManifest));
+    am->incr_aof_list = listCreate();
+    am->history_aof_list = listCreate();
+    listSetFreeMethod(am->incr_aof_list,aofListFree);
+    listSetDupMethod(am->incr_aof_list,aofListDup);
+    listSetFreeMethod(am->history_aof_list,aofListFree);
+    listSetDupMethod(am->history_aof_list,aofListDup);
+    return am;
+}
+
+/* Free the aofManifest structure (pointed to by am) and its embedded members. */
+void aofManifestFree(aofManifest *am) {
+    if (am->base_aof_info) aofInfoFree(am->base_aof_info);
+    if (am->incr_aof_list) listRelease(am->incr_aof_list);
+    if (am->history_aof_list) listRelease(am->history_aof_list);
+    zfree(am);
+}
+
+char *getAofManifestFileName() {
+    return sdscatprintf(sdsempty(),"%s%s",server.aof_filename,MANIFEST_NAME_SUFFIX);
 }
 
-/* Return the current size of the AOF rewrite buffer. */
-unsigned long aofRewriteBufferSize(void) {
+char *getTempAofManifestFileName() {
+    return sdscatprintf(sdsempty(),"%s%s%s",MANIFEST_TEM_NAME_PREFIX,server.aof_filename,MANIFEST_NAME_SUFFIX);;
+}
+
+/* Returns the string representation of aofManifest pointed to by am.
+ * 
+ * The string is multiple lines separated by'\n', and each line 
+ * represents an AOF file.
+ * 
+ * Each line contains 6 fields (they are separated by spaces).Among 
+ * them, the 0th, 2nd, and 4th field respectively represent the manifest 
+ * key, they are:
+ * fileName: AOF file name
+ * fileSeq: The serial number of the AOF file
+ * fileType: Types of AOF file, There are three types: b(BASE)、h(HISTORY)、i(INCR)
+ * 
+ * A possible line:
+ *    fileName appendonly.1.rdb fileSeq 1 fileType b
+ * 
+ * The BASE AOF information (if we have) will be placed on the first 
+ * line, followed by history type AOFs and finally is the INCR AOFs.
+ */
+#define AOF_INFO_FORMAT(info)                               \
+    sdscatprintf(buf,"%s %s %s %lld %s %c\n",               \
+                AOF_MANIFEST_KEY_FILE_NAME,info->file_name,     \
+                AOF_MANIFEST_KEY_FILE_SEQ,info->file_seq,       \
+                AOF_MANIFEST_KEY_FILE_TYPE,info->file_type)     \
```

Comment:
i'd rather add spaces between arguments in function calls.
i know redis is inconsistent in that (each file and function has its own style), but i think that's where we should aim for in new code.
```suggestion
                AOF_MANIFEST_KEY_FILE_NAME, info->file_name,     \
                AOF_MANIFEST_KEY_FILE_SEQ, info->file_seq,       \
                AOF_MANIFEST_KEY_FILE_TYPE, info->file_type)     \
```